src/Main.idr

```diff
@@ -35,21 +37,22 @@ mainRug
        ast <- fromFile fname
        putStrLn "# Finished Parsing"
 
-       res <- elab ast
+       (ty ** holes ** t) <- synth [] ast
+       let (metas ** inv) = initInvariant [] [] holes
+       let t = wscoped t inv
        putStrLn "# Finished Type Checking"
 
-       case res of
-         Holly h => do prettyHoles h
-                       exitSuccess
-         Closed tm
+       case metas of
+         (_ :: _) => do prettyMetas metas
+                        exitSuccess
+         []
```

Comment:
A fix for later, but it would be nice to hide away some of the details here back into the elaborator. Pattern matching of a result (i.e.`Holly` or `Closed`) keeps the logic a bit more cleaner.